src/engine/SpatialJoin.cpp

```diff
@@ -0,0 +1,295 @@
+#include "SpatialJoin.h"
+#include "util/AllocatorWithLimit.h"
+#include "util/MemorySize/MemorySize.h"
+#include "global/ValueId.h"
+#include "ValuesForTesting.h"
+#include "parser/ParsedQuery.h"
+#include "VariableToColumnMap.h"
+#include "engine/ExportQueryExecutionTrees.h"
+#include "util/GeoSparqlHelpers.h"
+
+SpatialJoin::SpatialJoin(QueryExecutionContext* qec, SparqlTriple triple,
+  std::optional<std::shared_ptr<QueryExecutionTree>> childLeft,
+  std::optional<std::shared_ptr<QueryExecutionTree>> childRight,
+  int maxDist)
+            : Operation(qec) {
+  this->triple = triple;
+  this->maxDist = maxDist;
```

Comment:
Call the internal data members `triple_` etc (trailing underscore)
and then simply use
`triple_ = triple` etc. (The `this` pointer is rarely used explicitly in C++).

Reply:
(many occurences).